contracts/token/ERC721/extensions/draft-ERC721Votes.sol

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.3.2 (token/ERC721/extensions/draft-ERC721Votes.sol)
+
+pragma solidity ^0.8.0;
+
+import "../ERC721.sol";
+import "../../../utils/Voting.sol";
+import "../../../utils/Counters.sol";
+import "../../../utils/math/Math.sol";
+import "../../../utils/Checkpoints.sol";
+import "../../../utils/math/SafeCast.sol";
+import "../../../utils/cryptography/ECDSA.sol";
+import "../../../utils/cryptography/draft-EIP712.sol";
+
+/**
+ * @dev Extension of ERC721 to support Compound-like voting and delegation. This version is more generic than Compound's,
+ * and supports token supply up to 2^224^ - 1, while COMP is limited to 2^96^ - 1.
+ *
+ * This extension keeps a history (checkpoints) of each account's vote power. Vote power can be delegated either
+ * by calling the {delegate} function directly, or by providing a signature to be used with {delegateBySig}. Voting
+ * power can be queried through the public accessors {getVotes} and {getPastVotes}.
+ *
+ * By default, token balance does not account for voting power. This makes transfers cheaper. The downside is that it
+ * requires users to delegate to themselves in order to activate checkpoints and have their voting power tracked.
+ * Enabling self-delegation can easily be done by overriding the {delegates} function. Keep in mind however that this
+ * will significantly increase the base gas cost of transfers.
+ *
+ * _Available since v4.5._
+ */
+abstract contract ERC721Votes is ERC721, EIP712 {
+    using Counters for Counters.Counter;
+    using Voting for Voting.Votes;
+
+    uint256 _totalVotingPower;
```

Comment:
What do we need that for, isn't that a duplicate of 
```
_votes._totalCheckpoints[_votes._totalCheckpoints.length].value
```
?

Reply:
My bad, I just realized this is an equivalent of totalSupply().

I'm not a big fan of this for two reasons:
- it is a duplicate of something present in ERC721Enumerable
- it is actually expensive to maintain
- it protects against something that, IMO, will never happen.

Unlike ERC20, ERC721 are minted one at a time, for it to come into action, we would need to function to be called 2**224 time. There is not enough energy in our universe for that to be remotely possible.

I suggest we drop this storage slot, and the corresponding checks. @frangio ?

Reply:
The first approach was to inherit from `ERC721Enumerable` but it added unnecesarry overhead, since that not the only thing in that contract, and that's why we needed to keep it as a new storage variable.

Reply:
But do we? It's only used for checking that minting if not going over 2**224, which is never going to happen in this univers